### PR TITLE
[BOUNTY] Yogstation Mouse Expansion pack, only 19,99$

### PR DESCRIPTION
**Pull request by alexkar598**
This PR has been funded by Hopek
This PR has been inspired by the following forums post: https://forums.yogstation.net/index.php?threads/aquir-ghost-role-mouse.20310/

You can now control mice that randomly spawns in maint, you can do so via the new Possess Mouse verb in the Ghost tab

Features:
- They can't chew wires (only player controlled)
- They hide under items now (only player controlled)
- They only have 15 health
- Night vision
- They can't pull items
- They can't push items
- They can pass under airlocks like drones
- They have their own language now, others can only understand it as Squeak? or Squeak!(its all in the microtone variations man)
- AI turrets now ignores mice
- They can eat stuff by control clicking it and waiting 3 seconds, during that time, they lose their ability to hide, they can eat trash,vomit and food that is not secured on tables
- Each food item eaten heals 5 damage except for cheese which does 10
- When eating an item while already healed, the remaining HP will go into a global counter that when it hits 35 spawns a new mouse in maint
- Mouse traps now hide under items, it also only does between 7 and 12 damage to player controlled mouse instead of an instant kill
- When a mouse eats cheese, they become THICC(bigger sprite), have doubled health and double speed for 3 minutes after which they become back normal
- Walking on the same tile as drinks or beakers not on tables will make the contents fall over

Pics ;)
I sure do hope my unsecured beakers dont get destroyed by a mouse in a suicide run!

The mouse eats it all!

Night vision mouse + airlock thing

Tactical mouse

Cheese and banana peels

The death of a "bad" guy :*(

### CODE REVIEW

**Comment by alexkar598** [comment]
Some of the gifs are too **THICC** to be loaded on github

**Comment by monster860** [comment]
This is the most important PR we've had in a very long time. This will completely change the nature of the game, and finally make us better than every other server ever.

**Comment by Ktlwjec1** [comment]
Absolutely unacceptable. This will completely fuck blobs even more than they already are, screw over revheads early and late game, contribute to the growing holoparasite problem, reinforce the emag-revolver meta, and destroy the viability of shadowlings. I don't know how you could even consider this change.

Did it ever occur to you how much this would affect traitors, abductors, changelings, nukeops, deathsquads, and megafauna? This game is a complex balancing act of interwoven checks and balances, maintained by some of the most highly trained and educated spaceman experts in the world, and this PR shows complete disregard for that haphazard yet workable tapestry which we have spent five years holding together with duct tape and bubblegum.

If by some miracle this is merged we may as well set the probability of all gamemodes to zero except extended so that the playerbase can at least eke out some meager entertainment from roleplay, since any semblance of balance will be utterly and completely destroyed.

All in all decent addition 👍

**Comment by Readystorm546** [comment]
Can we just have it so that when you eat cheese and grow bigger it makes it permanent and its all in effort to see who can become the biggest mouse.

**Comment by Darkstick** [comment]
This is making me very happy y es yes speed merge now

**Comment by Jarod1200** [comment]
Is the curator going to speak rat now?

**Comment by alexkar598** [comment]
Yes, he speaks fluent mouse

**Comment by Jarod1200** [comment]
Time to become onion the rat-mancer

**Comment by ThatGuyPhil** [comment]
*lizardpeople noises*

**Comment by swissloaf** [comment]
Call it a DLC

**Comment by Ratvar** [comment]
I do not like giving librarian literal ghost language. It is as if he could understand drones

**Comment by alexkar598** [comment]
I dont want to ruin for you, but he can understand drones

**Comment by Hopekz** [comment]
> I do not like giving librarian literal ghost language. It is as if he could understand drones

Curators are omni lingual which means they understand and speak all languages; including drone.

File: code/modules/language/mouse.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/datum/language/mouse
+	name = "Mouse"
+	desc = "A rather simple language spoken and understood by mice. It has been developped and shared by genetically advanced mice."
```
Comment (Ktlwjec1) [inline code/modules/language/mouse.dm]: developed

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -899,3 +900,66 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+		if(M.stat != CONSCIOUS) continue
+		if(M.key) continue
+		if(M in GLOB.player_list) continue
+		if(M.mind) continue
+		if(!is_station_level(M.z)) continue
+
+		possessible += M
+
+	if(!possessible.len)
+		to_chat(src, "<span class='warning'>There are currently no mice able to be possesed</span>")
```
Comment (Ktlwjec1) [inline code/modules/mob/dead/observer/observer.dm]: Excuse me where is the full stop at the end of this sentence. + possessed spelling

File: code/modules/mob/dead/observer/observer.dm [hunk 3]
```diff
@@ -899,3 +900,66 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+
+	possess_mouse(M)
+
+
+/mob/dead/observer/proc/possess_mouse(mob/living/simple_animal/mouse/M)
+	if(!M)
+		return FALSE		
+
+	if(!SSticker.HasRoundStarted())
+		to_chat(usr, "<span class='warning'>The round hasen't started yet!</span>")
```
Comment (Ktlwjec1) [inline code/modules/mob/dead/observer/observer.dm]: hasn't

File: code/modules/mob/dead/observer/observer.dm [hunk 4]
```diff
@@ -899,3 +900,66 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+	M.faction = list("neutral")
+	M.chew_probability = 0 //so they cant pull off a big brain play by ghosting somewhere or idk
+	M.layer = BELOW_OPEN_DOOR_LAYER //ENGAGE ADVANCED HIDING BRAIN FUNCTIONS
+	M.language_holder = new /datum/language_holder/mouse(M)
+	M.pass_flags |= PASSDOOR
+	M.sentience_act()
+	M.maxHealth = 15
+	M.health = M.maxHealth
+
+	to_chat(M , "<span class='warning'>You are now possessing a mouse \
```
Comment (Ktlwjec1) [inline code/modules/mob/dead/observer/observer.dm]: mouse full stop

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 5]
```diff
@@ -98,7 +128,104 @@
...
+	if(cheesed)
+		return
+	cheesed = TRUE
+	regen_health(15)
+	resize = 2
+	update_transform()
+	add_movespeed_modifier(MOVESPEED_ID_MOUSE_CHEESE, TRUE, 100, multiplicative_slowdown = -1)
+	maxHealth = 30
+	health = maxHealth
+	to_chat(src, "<span class='userdanger'>You have ate cheese! You are now stronger, bigger and faster!</span>")
```
Comment (Ktlwjec1) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Should be "You have eaten cheese!" or "You ate cheese!"

**Comment by ghost** [comment]
there goes the station's powernet

**Comment by alexkar598** [comment]
Player controlled mice cannot chew on power grid, the mice they create by eating trash however can, until they become player controlled

**Comment by alexkar598** [comment]
There is now a round end stat screen between medals and station goals(most of the time medals will be hidden and its going to show between antag and station goals). Screen only shows if atleast 1 trash was eaten

**Comment by Ratvar** [comment]
Drones were removed from massprinting roundstart and have special rules preventing them from chatting to people - and even then it was dumb. Will mice have drone-like rules to stop them from going <trator here>?

**Comment by alexkar598** [comment]
What can the mouse possibly do?

**Comment by Firewolf34** [comment]
This is fucking AWESOME!

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 6]
```diff
@@ -79,8 +108,15 @@
 					
 /mob/living/simple_animal/mouse/Move()
 	. = ..()
-	if(!stat)
+	if(!stat && !key)
 		eat_cheese()
+	if(key && stat == CONSCIOUS)
```
Comment (monster860) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: this and the above are saying `stat == CONSCIOUS` in two different ways

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 7]
```diff
@@ -98,7 +134,107 @@
 		qdel(O)
 	else
 		return ..()
-		
+
+/mob/living/simple_animal/mouse/attack_ghost(mob/dead/observer/user)
+	user.possess_mouse(src)
```
Comment (monster860) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:142]: Theres no confirmation dialog it seems - if I constantly click on fire alarms by accident, a ghost is gonna click on a mouse by accident and then there'll be a problem

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 8]
```diff
@@ -98,7 +134,107 @@
...
+	layer = BELOW_OPEN_DOOR_LAYER
+
+/mob/living/simple_animal/mouse/proc/can_eat(atom/A)
+	. = FALSE
+
+	if(eating)
+		return FALSE
+	if(is_type_in_list(A, GLOB.mouse_combestible))
+		return TRUE
+	if(istype(A, /obj/item/reagent_containers/food) && !(locate(/obj/structure/table) in get_turf(A)))
```
Comment (monster860) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:178]: Is it intentional that mouse_combestible bypasses table checks? also combestible isnt even a real word

**Review by monster860 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 7 above)
Comment (alexkar598) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:142]: prompt is done at https://github.com/yogstation13/Yogstation/pull/7838/files#diff-9c1590efced1526ca7e5cfdae3a5a19eR941

observer.dm L941

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 8 above)
Comment (alexkar598) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:178]: Yes it is, fixed combestible

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 6 above)
Comment (alexkar598) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm]: Changed the check to be more consistant

**Comment by deathslayer2481** [comment]
Oh God its merged

**Comment by 00aquir** [comment]
Food secured on tables.. mice can dump donuts out of the box from the table onto the floor.

**Comment by Titanslayer71** [comment]
Uh. Excuse me. We had a problem in (Round Id: 27966). A group of mice players were eating rice and brought the time dilation of the server to 75%. The admin then killed all the rats and dropped it back to 25%.

I was wondering if you can implement a way to curve population. Thanks.

You can watch the round over here. https://yogstation13.github.io/demo-viewer/?roundid=27966#30038;165.50;143.50;2;titanslayer71

It got crazy with the amount of rats eating...

**Pull request merged**